Wallet: Use SegWit by default

pinheadmz opened the pull request:
Simple change to the default value in `WalletDB` so all newly created wallets generate segregated witness addresses. Had to update several tests to comply.

braydonf commented:
Concept ACK. I think these is done most of the time anyways when creating new addresses.

braydonf commented:
> ...why the miner requires higher fees...

Similar, especially since it would seem the reverse would be true.

pinheadmz commented:
@braydonf @tynes see #666 
I never figured it out but recorded a lot of data about the scenario there, can look more into it next week.

nodar-chkuaselidze commented on test/http-test.js:
```diff
@@ -139,7 +140,7 @@ describe('HTTP', function() {
     assert(receive);
     assert.strictEqual(receive.name, 'default');
-    assert.strictEqual(receive.type, 'pubkeyhash');
+    assert.strictEqual(receive.type, 'witness');
```
I think we should have both test cases, should not be hard to have both legacy and witness wallet tests.

pinheadmz commented:
Rebased on master (`client` now included in bcoin repo).
Separated node from wallet HTTP tests and added a `for` to cover both legacy and segwit wallet operations.

tuxcanfly reviewed (approved):
Tested. LGTM :+1:
